pkg/cfn/builder/vpc_endpoint.go

```diff
@@ -123,16 +123,27 @@ func (e *VPCEndpointResourceSet) hasEndpoint(endpoint string) bool {
 	return false
 }
 
-// BuildVPCEndpointServices builds a slice of VPCEndpointServiceDetails for the specified endpoint names
-func BuildVPCEndpointServices(ec2API ec2iface.EC2API, region string, endpoints []string) ([]VPCEndpointServiceDetails, error) {
+var chinaPartitionServiceHasChinaPrefix = map[string]bool{
+	api.EndpointServiceEC2:            true,
+	api.EndpointServiceECRAPI:         true,
+	api.EndpointServiceECRDKR:         true,
+	api.EndpointServiceS3:             false,
+	api.EndpointServiceSTS:            true,
+	api.EndpointServiceCloudFormation: true,
+	api.EndpointServiceAutoscaling:    true,
+	api.EndpointServiceCloudWatch:     false,
+}
```

Comment:
Since only S3 and CloudWatch do not have a prefix, we can have a map called `chinaPartitionServicesWithoutPrefix` containing just the two services, and invert the condition in the code below.

Reply:
Ah I actually meant to add a check to make sure it's in the map and fail if not. I don't think we should assume either way.

Reply:
added, WDYT?

Reply:
That validation is already done in `ValidatePrivateCluster` before this code is run.

Reply:
I don't think so, I'm checking whether we have explicitly added the china prefix for that service, which is more of an assertion about the code.

Reply:
Ah, okay. This redundant check does require remembering to update two lists when a new endpoint service is added, and forgetting to update it here will break cluster creation for China regions.